docs/source/tutorials/vision/custom_task_tutorial.rst

```diff
@@ -0,0 +1,280 @@
+====================
+Custom Task Tutorial
+====================
+
+Computer vision is consists of different objectives models are trained for. The desired objective reflects
+on the structure of the data and the possible actions on it.
+
+In Deepchecks the first step before running checks is to create an implementation of `VisionData` which
+standardize the structure of the data for a given computer vision task.
+
+Some of the checks are agnostic to the specific task and will work on all of them, and some are tied to the
+specific details of a task and won't work for other tasks. When implementing your own task all of the
+task-agnostic checks will be able to run.
+
+Defining the Data & Model
+=========================
+
+First we will define a `Dataset` of COCO-128 segmentation task.
```

Comment:
maybe note that this is a setup stage and that the deepchecks specific code will be in the next section?